specification/saas/resource-manager/Microsoft.SaaS/preview/2018-03-01-beta/saas.json

```diff
@@ -0,0 +1,866 @@
+            }
+          },
+          "default": {
+            "description": "A response indicating an error.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get saas application": {
+            "$ref": "./examples/saasV1/SaaSGetApplications.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/providers/Microsoft.SaaS/saasresources/{resourceId}": {
+      "delete": {
+        "tags": [
+          "SaaSResources"
+        ],
+        "operationId": "SaaS_Delete",
+        "description": "Deletes the specified SaaS.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SaasResourceIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/deleteOptions"
+            },
+            "description": "Parameters supplied to delete saas operation."
+          }
+        ],
+        "responses": {
+          "204": {
+            "description": "Request successful."
+          },
+          "200": {
+            "description": "Request successful."
+          },
+          "default": {
+            "description": "A response indicating an error.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Delete SaaS resource": {
+            "$ref": "./examples/saasV2/SaasDelete.json"
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        }
+      },
+      "get": {
+        "tags": [
+          "SaaSResources"
+        ],
+        "operationId": "Saa_GetResource",
+        "description": "Gets information about the specified SaaS.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SaasResourceIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful. The request returns the resulting SaaS resource.",
+            "schema": {
+              "$ref": "#/definitions/SaasResource"
+            }
+          },
+          "default": {
+            "description": "A response indicating an error.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get SaaS resource": {
+            "$ref": "./examples/saasV2/SaasGet.json"
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.SaaS/saasresources": {
+      "put": {
```

Comment:
>put [](start = 7, length = 3)

Put on a resource collection is not allowed. See [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md).

Reply:
its a put on a single tenant level resource

Reply:
That is also compliant with the resource provider contract in ARM either as `saasresources` segment is in the place of the collection segment. You mentioned that this is an old API that you are just documenting. Can you confirm that this applies here as well?

---
In reply to: [529391536](https://github.com/Azure/azure-rest-api-specs/pull/11680#discussion_r529391536) [](ancestors = 529391536)

Reply:
yes, its an existing API that I'm documenting here, and that there are no changes in the service.